truss/remote/baseten/core.py

```diff
@@ -93,8 +93,17 @@ def create_chain(
         # if there is no chain already, the first deployment will
         # already be production, and only published deployments can
         # be promoted.
-        response = api.deploy_chain_deployment(chain_id, chainlets, promote)
+        try:
+            response = api.deploy_chain_deployment(chain_id, chainlets, environment)
+        except ApiError as e:
+            if "Environment matching query does not exist" in e.message:
```

Comment:
can we move this string into a constant?

We don't have to fix this now, but I wish this could be passed via a code vs. having to do this exact string matching

Reply:
added a check for `e.graphql_error_code  == BasetenApi.GraphQLErrorCodes.RESOURCE_NOT_FOUND.value` instead of the string matching